mapiproxy/servers/default/emsmdb/oxcfxics.c

```diff
@@ -3428,44 +3526,11 @@ static enum MAPISTATUS oxcfxics_fill_transfer_state_arrays(TALLOC_CTX *mem_ctx,
 	struct GUID			replica_guid;
 	enum mapistore_error		ret;
 	enum MAPISTATUS			retval;
+	struct idset			*involved_fmids;
 	
 	local_mem_ctx = talloc_new(NULL);
 	OPENCHANGE_RETVAL_IF(local_mem_ctx == NULL, MAPI_E_NOT_ENOUGH_MEMORY, NULL);
 
-	/* Query the amount of rows and update sync_data structure */
-	count_query_props = talloc_zero(local_mem_ctx, struct SPropTagArray);
-	OPENCHANGE_RETVAL_IF(!count_query_props, MAPI_E_NOT_ENOUGH_MEMORY, local_mem_ctx);
-
-	count_query_props->cValues = 1;
-	count_query_props->aulPropTag = talloc_zero(count_query_props, enum MAPITAGS);
-	OPENCHANGE_RETVAL_IF(!count_query_props->aulPropTag, MAPI_E_NOT_ENOUGH_MEMORY,
-			     local_mem_ctx);
-	switch (sync_data->table_type) {
-	case MAPISTORE_FOLDER_TABLE:
-		count_query_props->aulPropTag[0] = PidTagFolderChildCount;
-		break;
-	case MAPISTORE_MESSAGE_TABLE:
-		count_query_props->aulPropTag[0] = PidTagContentCount;
-		break;
-	case MAPISTORE_FAI_TABLE:
-		count_query_props->aulPropTag[0] = PidTagAssociatedContentCount;
-		break;
-	default:
-		OC_DEBUG(1, "Wrong table type to transfer state: %d", sync_data->table_type);
-		talloc_free(local_mem_ctx);
-		return MAPI_E_INVALID_PARAMETER;
-	}
-	data_pointers = emsmdbp_object_get_properties(local_mem_ctx, emsmdbp_ctx, folder_object, count_query_props, (enum MAPISTATUS **) &retvals);
-	if (data_pointers && !retvals[0]) {
-		nr_eid = *(uint32_t *) data_pointers[0];
-		/* No content, return */
-		OPENCHANGE_RETVAL_IF(!nr_eid, MAPI_E_SUCCESS, local_mem_ctx);
-	} else {
-		OC_DEBUG(1, "ERROR: could not retrieve number of rows in table\n");
-		talloc_free(local_mem_ctx);
-		return MAPI_E_CALL_FAILED;
-	}
-
```

Comment:
It wasn't needed because it seems an early optimisations on previous implementation to avoid going through an empty table. This is not necessary and it reduces a lot the readability of the procedure IHMO. If there is no content in the end, no new change numbers neither identifiers will be added to the returned ids. What do you think?